Add Teal language support

mtdowling opened the pull request:
See https://teal-language.org

## Description
This PR adds support for the Teal, a statically-typed dialect of Lua.

## Checklist:
- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      -  https://github.com/search?q=%22--+%22+%28path%3A*.tl+OR+path%3A*..d.tl%29&type=code&ref=advsearch&p=1 (2.8K files)
  - [X] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/teal-language/cyan/blob/main/src/cyan/commands/build.tl
      - https://github.com/mtdowling/emitter.tl/blob/main/src/emitter.tl
    - Sample license(s): MIT
  - [X] I have included a syntax highlighting grammar: https://github.com/teal-language/vscode-teal
  - [X] I have added a color
    - Hex value: `#00B1BC`
    - Rationale: Teal is teal
  - [X] I have updated the heuristics to distinguish my language from others using the same extension.

lildude reviewed (changes requested):
See inline comment.

We also explicitly state we want real world samples (contrived samples are not good ways to train the classifier) and will reject PRs with "Hello world" samples… your `types.tl` looks remarkable like a "Hello world" sample.

Please replace your samples with _real_ samples and update the PR template to link to the original sources (not Linguist or your fork).

When you added the grammar, the `script/add-grammar` should have also generated a cached version of the grammar's license. This needs to be added to this PR too.

And finally…

> Search results for each extension:
>
>    https://github.com/search?q=filename%3Atlconfig.lua+NOT+is%3Afork&type=code (359 repositories)

I'm not sure how you got the number when your search query is invalid. Correcting it to search for just that filename returns only 87 files which is a very very long way from meeting our usage requirements. I recommend you update your search to search for the `.tl` extension and use your heuristic (search accepts regex) to narrow the results down to just this language.

mtdowling commented:
> We also explicitly state we want real world samples (contrived samples are not good ways to train the classifier) and will reject PRs with "Hello world" samples… your types.tl looks remarkable like a "Hello world" sample.

I've updated the sample to show code from a two public Teal projects. The CONTRIBUTING.md file has some wording that made me think samples made just for the PR were ok, but I've removed them and used real world code.

> This needs to be added to this PR too.

Done

> I'm not sure how you got the number when your search query is invalid

Sorry! Copy paste error. I've updated the search in the PR. I think [this search](https://github.com/search?q=%22--+%22+%28path%3A*.tl+OR+path%3A*..d.tl%29&type=code&ref=advsearch&p=1) in more representative, and show 2.8K files, though I don't know how to slice that into unique repositories.

lildude commented on vendor/licenses/git_submodule/vscode-teal.dep.yml:
```diff
@@ -0,0 +1,46 @@
+version: f4b0541c42a548ab82e34a2bab2dc5fa2cc9454b
+type: git_submodule
+homepage: https://github.com/teal-language/vscode-teal.git
+license: other
```
```suggestion
license: mit
```
